packages/lodestar/test/unit/chain/genesis/genesis.test.ts

```diff
@@ -20,77 +21,101 @@ describe("genesis builder", function () {
+      blocks.push({
+        number: i,
+        timestamp: schlesiConfig.params.MIN_GENESIS_TIME + i,
+        hash: `0x${String(i).padStart(64, "0")}`,
+      });
     }
 
-    eth1Stub = sandbox.createStubInstance(EthersEth1Notifier);
-    dbStub = new StubbedBeaconDb(sandbox);
-    loggerStub = sandbox.createStubInstance(WinstonLogger);
-    genesisBuilder = new GenesisBuilder(schlesiConfig, {
-      eth1: eth1Stub,
-      db: dbStub,
-      logger: loggerStub,
-    });
-  });
-
-  afterEach(() => {
-    sandbox.restore();
-  });
+    return {events, keypairs, blocks};
+  }
 
   it("should build genesis state", async () => {
-    const eth1Source = pushable<Eth1EventsBlock>();
-    eth1Stub.getEth1BlockAndDepositEventsSource.resolves(eth1Source);
-
-    const statePromise = genesisBuilder.waitForGenesis();
-    // push 2 first events at the same time first
-    eth1Source.push({events: [events[0], events[1]]});
-    for (let i = 2; i < schlesiConfig.params.MIN_GENESIS_ACTIVE_VALIDATOR_COUNT - 1; i++) {
-      eth1Source.push({events: [events[i]]});
-    }
-    const lastEventIndex = schlesiConfig.params.MIN_GENESIS_ACTIVE_VALIDATOR_COUNT - 1;
-    const block = {
-      number: lastEventIndex,
-      timestamp: Math.floor(Date.now() / 1000),
-      hash: "0x54b9f905f15634d966690bd362381cfd7a28362d683f8d1616aa478b575152f8",
-    } as ethers.providers.Block;
-    eth1Source.push({events: [events[lastEventIndex]], block});
+    const {blocks, events} = generateGenesisBuilderMockData();
+
+    const eth1Provider: IEth1Provider = {
+      deployBlock: events[0].blockNumber,
+      getBlockNumber: async () => 2000,
+      getBlock: async (number) => blocks[number],
+      getDepositEvents: async (fromBlock, toBlock) =>
+        events.filter((e) => e.blockNumber >= fromBlock && e.blockNumber <= (toBlock || fromBlock)),
+      validateContract: async () => {
+        return;
+      },
+    };
 
-    const state = await statePromise;
-    expect(state.validators.length).to.be.equal(4);
+    const genesisBuilder = new GenesisBuilder(schlesiConfig, {
+      eth1Provider,
+      logger: new WinstonLogger(),
+      MAX_BLOCKS_PER_POLL: 1,
+    });
+
+    const {state} = await genesisBuilder.waitForGenesis();
 
+    expect(state.validators.length).to.be.equal(schlesiConfig.params.MIN_GENESIS_ACTIVE_VALIDATOR_COUNT);
     expect(toHexString(state.eth1Data.blockHash)).to.be.equal(
-      "0x54b9f905f15634d966690bd362381cfd7a28362d683f8d1616aa478b575152f8"
+      blocks[schlesiConfig.params.MIN_GENESIS_ACTIVE_VALIDATOR_COUNT - 1].hash
     );
   });
 
-  function generateDeposit(index: ValidatorIndex): DepositData {
-    const domain = computeDomain(config, DomainType.DEPOSIT);
-    const depositMessage = {
-      pubkey: keypairs[index].publicKey.toBytesCompressed(),
-      withdrawalCredentials: Buffer.alloc(32, index),
-      amount: BigInt(32 * 1000000000000000000),
+  it("should abort building genesis state", async () => {
```

Comment:
@mpetrunic I'm testing stream aborting through this test. Should I test them directly anyway? stream abort tests are very similar to the one below anyway.

Reply:
think this is fine